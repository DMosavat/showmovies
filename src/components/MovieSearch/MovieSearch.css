
.MovieSearch-input{
    width:300px;
    margin-right:10px
}

.MovieSearch-Card{
    background-color: #2A2323
}

.MovieSearch-button{
    color: #FFF;
    font-size: 14px
}