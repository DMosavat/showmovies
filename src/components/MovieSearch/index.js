import React from 'react';
import { Card } from 'react-bootstrap'

import './MovieSearch.css'

function MovieSearch(props){
    
   let { Search, fetchMovieID, showFavoriteMovie, mName } = props

   return(
        <Card>
            <Card.Body>
            <div className="row">
                <input 
                type="text" 
                className={["MovieSearch-input"]}
                placeholder="Search movie name ..."
                name="mName"
                value={mName}
                onChange={(e)=>Search(e)} />

                <div >
                    <button className="btn btn-link" 
                        onClick={(e)=>{e.preventDefault(); fetchMovieID(0) }}
                    > New Film 
                    </button>

                    <button className="btn btn-link" 
                        onClick={(e)=>{e.preventDefault(); fetchMovieID(1) }}
                    > Top Film
                    </button>

                    <button className="btn btn-link" 
                        onClick={(e)=>{e.preventDefault(); showFavoriteMovie() }}
                    > Favorites
                    </button>
                </div>
            </div>

            </Card.Body>
        </Card>
   )
}

export default MovieSearch;
