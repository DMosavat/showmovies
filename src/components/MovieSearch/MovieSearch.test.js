import React from 'react';
import { shallow } from 'enzyme';
import MovieSearch from './index'

describe('MovieSearch Test', () => {

    describe('<MovieSearch />', () => {

        it('should render the form', () => {
            const wrapper = shallow(<MovieSearch />);
            expect(wrapper.html()).toMatchSnapshot();
        })
    
    })
})