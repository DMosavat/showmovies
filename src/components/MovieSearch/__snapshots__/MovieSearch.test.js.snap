// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSearch Test <MovieSearch /> should render the form 1`] = `"<div class=\\"MovieSearch-Card card\\"><div class=\\"card-body\\"><div class=\\"row\\"><input type=\\"text\\" class=\\"MovieSearch-input\\" placeholder=\\"Search movie name ...\\" name=\\"mName\\"/><div><button class=\\"btn btn-link MovieSearch-button\\"> New Film</button><button class=\\"btn btn-link MovieSearch-button\\"> Top Film</button><button class=\\"btn btn-link MovieSearch-button\\"> Favorites</button></div></div></div></div>"`;
